src/rgw/rgw_reshard.cc

```diff
@@ -628,6 +628,13 @@ int RGWBucketReshard::do_reshard(int num_shards,
 	rgw_bucket_category_stats stats;
 	bool account = entry.get_info(&cls_key, &category, &stats);
 	rgw_obj_key key(cls_key);
+        if (entry.type == BIIndexType::OLH && key.name.empty()) {
```

Comment:
maybe this should be `key.empty()`, equivalent afaict.

Reply:
Yes, the rgw_obj_key struct has three fields -- name, instance, and namespace. But it's empty() function only looks at name. So it's equivalent. Probably cleaner to use key.empty(), though.